discoverer/src/main/java/eu/seaclouds/platform/discoverer/core/Discoverer.java

```diff
@@ -17,39 +17,51 @@
 
 package eu.seaclouds.platform.discoverer.core;
 
-import eu.seaclouds.platform.discoverer.api.*;
+import com.mongodb.MongoClient;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import eu.seaclouds.platform.discoverer.crawler.CloudHarmonySPECint;
 import eu.seaclouds.platform.discoverer.crawler.CrawlerManager;
-import io.dropwizard.Application;
-import io.dropwizard.setup.Environment;
+import org.bson.Document;
 
+import java.io.*;
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Date;
 
 
-public class Discoverer extends Application<DiscovererConfiguration> {
+public class Discoverer {
     /* singleton */
-    private static final Discoverer singleInstance = new Discoverer();
     private ArrayList<String> offeringNodeTemplates = new ArrayList<>();
     private boolean refreshing = false;
     private ArrayList<String> activeCrawlers;
 
+    private String databaseName = "DrACO_DB";
+    private String collectionName = "offerings";
+
     public int totalCrawledOfferings;
     public int crawledTimes;
     public Date lastCrawl;
 
-    public static Discoverer instance() { return singleInstance; }
+    public Discoverer(MongoClient mongoClient, ArrayList<String> activeCrawlers) {
+        this.initializeResources();
 
-    /* vars */
-    public OfferingManager offeringManager;
+        MongoDatabase db = mongoClient.getDatabase(this.databaseName);
+        MongoCollection<Document> coll = db.getCollection(this.collectionName);
```

Comment:
The logic is unchanged, crawlers add offerings into the collection (instead of creating files on file system) and, in case CH or PaaSify are not available, we can import an image of the collection.